dataland-frontend/src/components/resources/dataTable/conversion/MultiSelectValueGetterFactory.ts

```diff
@@ -19,37 +20,54 @@ export function multiSelectValueGetterFactory(
   field: Field,
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
 ): (dataset: any) => AvailableMLDTDisplayObjectTypes {
-  const nameMap = new Map<string, string>();
-  for (const option of field.options ?? []) {
-    nameMap.set(option.value, option.label);
-  }
-
   return (dataset) => {
-    const selectionValue = getFieldValueFromFrameworkDataset(path, dataset) as Array<string>;
-    if (!selectionValue || selectionValue.length == 0) {
-      return {
-        displayComponentName: MLDTDisplayComponentName.StringDisplayComponent,
-        displayValue: "",
-      };
-    } else {
-      return <MLDTDisplayObject<MLDTDisplayComponentName.ModalLinkDisplayComponent>>{
-        displayComponentName: MLDTDisplayComponentName.ModalLinkDisplayComponent,
-        displayValue: {
-          label: `Show ${selectionValue.length} value${selectionValue.length > 1 ? "s" : ""}`,
-          modalComponent: MultiSelectModal,
-          modalOptions: {
-            props: {
-              header: field.label,
-              modal: true,
-              dismissableMask: true,
-            },
-            data: {
-              label: field.label,
-              values: selectionValue.map((it) => nameMap.get(it)),
-            },
-          },
-        },
-      };
+    let fieldValueFromFrameworkDataset = getFieldValueFromFrameworkDataset(path, dataset) as Array<string>;
+    if (fieldValueFromFrameworkDataset == undefined) {
+      fieldValueFromFrameworkDataset = [];
+    }
+
+    const expectedDropDownOptions = field.options ?? [];
+    const technicalNameToLabelMapping = new Map<string, string>();
```

Comment:
Why do we need this?

Reply:
Because you want to check the validity of the options from the dataset and also use the correct "humand-readable" text for the display.
`technicalNameToLabelMapping` contains all the technical names, that are expected to potentially be included in the dataset field, and the correct human-readable display names for them:

Example:
Imagine `technicalNameToLabelMapping` to be 
``{Employed: "employed", Unemployed: "unemployed", PartTimeEmployed: "part-time employed", FullTimeEmployed: "full-time employed", SelfEmployed: "self-employed"}``

Now if some field `employmentStatus` gives this function a list with data like this; ["PartTimeEmployed", "SelfEmployed"], it knows the humand-readable names to dsiplay.

Reply:
Ok, thank you for the explanation